File: 37-Python/Socio.py
class Socio:
    def __init__(self, codiceSocio, nomeCognome, eta, categoria) -> None:
        self._codiceSocio = codiceSocio
        self._nomeCognome = nomeCognome
        self._eta = eta
        self._categoria = categoria
        self._prenotazioni = []

    def addPrenotazioni(self, prenotazione):
        self._prenotazioni.append(prenotazione)
    
    def __str__(self) -> str:
        return str(self._codiceSocio) +"\t"+ self._nomeCognome +"\t"+ str(self._eta) +"\t"+ str(self._categoria) +"\t"+ str([str(prenotazione) for prenotazione in self._prenotazioni])

File: 37-Java/src/Socio.java
import java.util.*;

public class Socio {
    private int codiceSocio, eta, categoria;
    private String nomeCognome;
    private List<Prenotazione> prenotazioni;

    public Socio(int codiceSocio, int eta, int categoria, String nomeCognome) {
        this.codiceSocio = codiceSocio;
        this.eta = eta;
        this.categoria = categoria;
        this.nomeCognome = nomeCognome;
        prenotazioni = new LinkedList<Prenotazione>();
    }

    public void addPrenotazione(Prenotazione p){
        prenotazioni.add(p);
    }

    public String toString(){
        return codiceSocio +"\t"+ nomeCognome +"\t"+ eta +"\t"+ categoria +"\t"+ prenotazioni;
    }
}
